### ISSUE TITLE
Unicode characters corrupt searchindex.js

### ISSUE THREAD

**Issue by ostrokach** [sphinx-doc/sphinx#2296]
If you have words with unicode characters in your *_.rst_ files, those words get inserted into the `searchindex.js` file, which makes javascript throw an error when searching: _"SyntaxError: Unexpected token ILLEGAL"_.

To reproduce, create an _index.rst_ file with a single line:

```
Kirill MÃ¼ller
```

Run `make html` to create the following _searchindex.js_ file:

```
Search.setIndex({envversion:47,filenames:["index"],objects:{},objnames:{},objtypes:{},terms:{kiril:0,mã¼ller:0},titles:["&lt;no title&gt;"],titleterms:{}})
```

Load the docs and attempt to search, causing an error: 

```
SyntaxError: Unexpected token ILLEGAL
#1 http://localhost:8000/searchindex.js:1:104
```

This can be fixed by adding the following checks to the `sphinx/search/__init__.py` file (probably would not work with Python 2):

```
366         for word in visitor.found_title_words:
367             word = stem(word)
368             # Make sure the word can be encoded as ascii (needed by javascript)
369             try:
370                 word.encode('ascii')
371             except UnicodeEncodeError:
372                 continue
373             if _filter(word):
374                 self._title_mapping.setdefault(word, set()).add(filename)
375         
376         for word in visitor.found_words:
377             word = stem(word)
378             # Make sure the word can be encoded as ascii (needed by javascript)
379             try:
380                 word.encode('ascii')
381             except UnicodeEncodeError:
382                 continue
383             if word not in self._title_mapping and _filter(word):
384                 self._mapping.setdefault(word, set()).add(filename)
```

**Comment by ostrokach** [comment]
Also, this is only an issue in Python 3.

In Python 2, `MÃ¼ller` gets converted into `m\u00e3\u00bcller` in the `searchindex.js` file, and everything works as expected.

**Comment by tk0miya** [comment]
It seems this bug was already fixed with recently release (maybe #2766).
So I close this now.
Please reopen if it is not fixed at latest release.

Thanks,
